File: combinatorics/resource/ST/project_5/bai_toan_4/general_graph/amap_to_al.cpp
#include <iostream>
#include <vector>
using namespace std;

int main() {
    int n, num_keys;
    cout << "Nhap so dinh: ";
    cin >> n;
    cout << "Nhap so dinh co trong AMap: ";
    cin >> num_keys;

    // Khoi tao danh sach ke
    vector<vector<int>> adjList(n);

    cout << "Nhap AMap (u, so dinh ke, cac cap v w):\n";
    for (int i = 0; i < num_keys; ++i) {
        int u, k;
        cin >> u >> k;
        for (int j = 0; j < k; ++j) {
            int v, w;
            cin >> v >> w; // Bo qua w
            adjList[u].push_back(v);
        }
    }

    cout << "Danh sach ke (AL):\n";
    for (int u = 0; u < n; ++u) {
        cout << u << ": ";
        for (int v : adjList[u]) {
            cout << v << " ";
        }
        cout << "\n";
    }

    return 0;
}

File: combinatorics/resource/ST/project_5/bai_toan_4/general_graph/amap_to_al.py
n = int(input("Nhap so dinh: "))
num_keys = int(input("Nhap so dinh co trong AMap: "))

# Khoi tao danh sach ke voi n danh sach con
adj_list = [[] for _ in range(n)]

print("Nhap AMap (u, so dinh ke, cac cap v w):")
for _ in range(num_keys):
    parts = input().split()
    u = int(parts[0])
    k = int(parts[1])
    for i in range(k):
        v = int(parts[2 + 2 * i])
        # Bo qua trong so w = parts[3 + 2 * i]
        adj_list[u].append(v)

# In danh sach ke
print("Danh sach ke (AL):")
for u in range(n):
    print(f"{u}:", adj_list[u])